app/controllers/application_controller.rb

```diff
@@ -32,6 +32,11 @@ def mnemosyne_trace
     end
   end
 
+  def login_from_authentication_token
+    token = AuthenticationToken.find_by(shared_secret: params[:token])
+    auto_login(token.user) if token&.user && token&.expire_at&.future?
```

Comment:
The scenario with and expired is not tested, is it?

Reply:
Fine for me. As commented on the changes, the scenarios testing the `future?` look a bit suscpicious.
Other than that, I'm very happy with the result.

Reply:
That's true, we also discussed the tests with the `future?` earlier today. As a consequence, the feature spec were added to test the actual behaviour for expired tokens.